Software/lib/GNSE_HAL/GNSE_hal.h

```diff
@@ -42,10 +42,21 @@ typedef struct GNSE_HAL_Ctx_s
     bool buzzer_init;
 } GNSE_HAL_Ctx_t;
 
+/**
+ * GNSE HAL return types
+ */
+typedef enum
+{
+    GNSE_HAL_OP_SUCCESS = 0,
+    GNSE_HAL_OP_FAIL = 1,
+} GNSE_HAL_op_result_t;
+
 void GNSE_HAL_Init(GNSE_HAL_Ctx_t gnse_inits);
 void GNSE_HAL_DeInit(GNSE_HAL_Ctx_t gnse_deinits);
 void GNSE_HAL_Internal_Sensors_Init(void);
 void GNSE_HAL_Internal_Sensors_DeInit(void);
+void GNSE_Error_Handler(void);
```

Comment:
Also minor nitpick, can you change the order of declarations/definition so that error handler is the last function.
It should be 
GNSE_HAL_SysClk_Init
GNSE_HAL_Error_Handler

in both the GNSE_hal.c and GNSE_hal.h